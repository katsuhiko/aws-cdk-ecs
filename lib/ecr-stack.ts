import { Construct, Stack, StackProps } from '@aws-cdk/core';
import { Repository } from '@aws-cdk/aws-ecr';

export class EcrStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    new Repository(this, 'WebRepo', {
      repositoryName: 'demo-cdk-web',
      imageScanOnPush: true,
      lifecycleRules: [
        { description: 'Delete old images', maxImageCount: 5 },
      ],
    });

    new Repository(this, 'AppRepo', {
      repositoryName: 'demo-cdk-app',
      imageScanOnPush: true,
      lifecycleRules: [
        { description: 'Delete old images', maxImageCount: 5 },
      ],
    });
  }
}
