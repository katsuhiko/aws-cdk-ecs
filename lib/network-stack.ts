import { Construct, Stack, StackProps } from '@aws-cdk/core';
import { SubnetType, Vpc } from '@aws-cdk/aws-ec2';

export class NetworkStack extends Stack {

  public readonly vpc: Vpc

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    this.vpc = new Vpc(this, 'VPC', {
      maxAzs: 2,
      subnetConfiguration: [
        { name: 'Frontend', subnetType: SubnetType.PUBLIC, cidrMask: 24 },
        { name: 'Application', subnetType: SubnetType.PUBLIC, cidrMask: 24 },
        { name: 'Datastore', subnetType: SubnetType.ISOLATED, cidrMask: 24 },
      ],
    });
  }
}
