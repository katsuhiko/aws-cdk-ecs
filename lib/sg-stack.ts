import { Construct, Stack, StackProps, Tags } from '@aws-cdk/core';
import { IVpc, Peer, Port, SecurityGroup } from '@aws-cdk/aws-ec2';

interface SgStackProps extends StackProps {
  vpc: IVpc
}

export class SgStack extends Stack {

  public readonly bastionSg: SecurityGroup;
  public readonly albSg: SecurityGroup;
  public readonly webSg: SecurityGroup;
  public readonly rdsSg: SecurityGroup;

  constructor(scope: Construct, id: string, props: SgStackProps) {
    super(scope, id, props);

    this.bastionSg = new SecurityGroup(this, 'BastionSg', {
      vpc: props.vpc,
    });
    Tags.of(this.bastionSg).add('Name', 'SgStack/BastionSg');

    this.albSg = new SecurityGroup(this, 'AlbSg', {
      vpc: props.vpc,
    });
    Tags.of(this.albSg).add('Name', 'SgStack/AlbSg');
    this.albSg.addIngressRule(Peer.anyIpv4(), Port.tcp(80));
    this.albSg.addIngressRule(Peer.anyIpv4(), Port.tcp(443));

    this.webSg = new SecurityGroup(this, 'WebSg', {
      vpc: props.vpc,
    });
    Tags.of(this.webSg).add('Name', 'SgStack/WebSg');
    this.webSg.addIngressRule(this.albSg, Port.tcp(80));

    this.rdsSg = new SecurityGroup(this, 'RdsSg', {
      vpc: props.vpc,
    });
    Tags.of(this.rdsSg).add('Name', 'SgStack/RdsSg');
    this.rdsSg.addIngressRule(this.bastionSg, Port.tcp(3306));
    this.rdsSg.addIngressRule(this.webSg, Port.tcp(3306));
  }
}
