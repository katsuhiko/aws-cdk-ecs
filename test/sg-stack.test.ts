import { App, Stack } from '@aws-cdk/core';
import { SynthUtils } from '@aws-cdk/assert';
import { Vpc } from '@aws-cdk/aws-ec2';
import { SgStack } from '../lib/sg-stack';

test('SG Stack', () => {
  const app = new App();

  const givenStack = new Stack(app, "GivenStack");
  const vpc = new Vpc(givenStack, 'TestVPC');

  const stack = new SgStack(app, 'TestStack', {
    vpc: vpc
  });

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
