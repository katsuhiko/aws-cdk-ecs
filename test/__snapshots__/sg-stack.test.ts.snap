// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SG Stack 1`] = `
Object {
  "Resources": Object {
    "AlbSg1155C1BE": Object {
      "Properties": Object {
        "GroupDescription": "TestStack/AlbSg",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "from 0.0.0.0/0:80",
            "FromPort": 80,
            "IpProtocol": "tcp",
            "ToPort": 80,
          },
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "from 0.0.0.0/0:443",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/AlbSg",
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "GivenStack:ExportsOutputRefTestVPCD26570D8E54DE509",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "BastionSg3D6BE75A": Object {
      "Properties": Object {
        "GroupDescription": "TestStack/BastionSg",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/BastionSg",
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "GivenStack:ExportsOutputRefTestVPCD26570D8E54DE509",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "RdsSg7F9F43B6": Object {
      "Properties": Object {
        "GroupDescription": "TestStack/RdsSg",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/RdsSg",
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "GivenStack:ExportsOutputRefTestVPCD26570D8E54DE509",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "RdsSgfromTestStackBastionSg48D7AC253306182B884C": Object {
      "Properties": Object {
        "Description": "from TestStackBastionSg48D7AC25:3306",
        "FromPort": 3306,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "RdsSg7F9F43B6",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "BastionSg3D6BE75A",
            "GroupId",
          ],
        },
        "ToPort": 3306,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "RdsSgfromTestStackWebSgC437B1633306FBB6D128": Object {
      "Properties": Object {
        "Description": "from TestStackWebSgC437B163:3306",
        "FromPort": 3306,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "RdsSg7F9F43B6",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "WebSg75AA54D5",
            "GroupId",
          ],
        },
        "ToPort": 3306,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "WebSg75AA54D5": Object {
      "Properties": Object {
        "GroupDescription": "TestStack/WebSg",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/WebSg",
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "GivenStack:ExportsOutputRefTestVPCD26570D8E54DE509",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "WebSgfromTestStackAlbSgF3BB9BFC80BD1C8FE3": Object {
      "Properties": Object {
        "Description": "from TestStackAlbSgF3BB9BFC:80",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "WebSg75AA54D5",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "AlbSg1155C1BE",
            "GroupId",
          ],
        },
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
  },
}
`;
